### ISSUE TITLE
valid-jsdoc for nested lambda functions expects both inner and outer params to be documented in the outer function

### ISSUE THREAD

**Issue by darthwalsh** [eslint/eslint#11468]
**Tell us about your environment**

* **ESLint Version:** v5.15.0
* **Node Version:** v11.5.0
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
    "env": {
        "browser": true,
        "es6": true
    },
    "extends": "google",
    "globals": {
        "Atomics": "readonly",
        "SharedArrayBuffer": "readonly"
    },
    "parserOptions": {
        "ecmaVersion": 2018
    },
    "rules": {
        "linebreak-style": "off"
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const Test = /** @param {any} player */ (player) => {
  const inner = /** @param {any} other */ (other) => {
    other.something();
  };

  inner(player);
};

module.exports.default = Test;

```

```bash
npx eslint test.js
```

**What did you expect to happen?**
No errors

**What actually happened? Please include the actual, raw output from ESLint.**
```
c:\code\temp\eslintbug>npx eslint test.js

c:\code\temp\eslintbug\test.js
  1:18  error  Expected JSDoc for 'other' but found 'player'  valid-jsdoc

✖ 1 problem (1 error, 0 warnings)
```

Weirdly, if I change the jsdoc to `@param {any} other` then I get an error "Expected JSDoc for 'player' but found 'other'".

And if I change the jsdoc to `@param {any} abc` I get an error that both `other` and `player` are missing
```
  1:18  error  Expected JSDoc for 'other' but found 'abc'   valid-jsdoc
  1:18  error  Expected JSDoc for 'player' but found 'abc'  valid-jsdoc
```

**Comment by not-an-aardvark** [comment]
Hi, thanks for the report.

The `valid-jsdoc` rule is [deprecated](https://eslint.org/blog/2018/11/jsdoc-end-of-life). In accordance with our [deprecation policy](https://eslint.org/docs/user-guide/rule-deprecation), we're no longer fixing any bugs in that rule. We recommend using something like [`eslint-plugin-jsdoc`](https://github.com/gajus/eslint-plugin-jsdoc) instead.
